pkg/quarterdeck/apikeys_test.go

```diff
@@ -204,6 +202,60 @@ func (s *quarterdeckTestSuite) TestCannotCreateAPIKeyInUnownedProject() {
 	s.CheckError(err, 400, "field restricted for request: org_id")
 }
 
+func (s *quarterdeckTestSuite) TestCannotCreateAPIKeyWithoutPermissions() {
+	require := s.Require()
+	defer s.ResetDatabase()
+
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	// Creating an API Key requires the apikeys:edit permission
+	// But cannot create an apikey with permissions that the user does not have.
+	claims := &tokens.Claims{
+		RegisteredClaims: jwt.RegisteredClaims{
+			Subject: "01GKHJSK7CZW0W282ZN3E9W86Z",
+		},
+		Name:        "Jannel P. Hudson",
+		Email:       "jannel@example.com",
+		OrgID:       "01GKHJRF01YXHZ51YMMKV3RCMK",
+		Permissions: []string{perms.EditAPIKeys},
+	}
+	ctx = s.AuthContext(ctx, claims)
+
+	// User should not be able to create an APIKey in a project not owned by that org.
+	req := &api.APIKey{
+		Name:        "Sneaky Key",
+		Source:      "Hacker",
+		ProjectID:   ulid.MustParse("01GQ7P8DNR9MR64RJR9D64FFNT"),
+		Permissions: []string{"topics:create", "topics:destroy", "topics:edit"},
+	}
+
+	_, err := s.client.APIKeyCreate(ctx, req)
+	s.CheckError(err, 400, "invalid permissions requested for apikey")
+
+	// Can create API key if the user has the permissions
+	claims.Permissions = []string{perms.EditAPIKeys, perms.CreateTopics, perms.DestroyTopics, perms.EditTopics}
+	ctx = s.AuthContext(ctx, claims)
+
+	_, err = s.client.APIKeyCreate(ctx, req)
+	require.NoError(err, "should have been able to create api key with permissions")
+
+	// Cannot create API key with permission that is not allowed for api keys
+	req.Permissions = []string{"publisher", "subscriber", "organizations:delete", "topics:create"}
+	_, err = s.client.APIKeyCreate(ctx, req)
+	s.CheckError(err, 400, "validation error: invalid permission specified for apikey")
+
+	// Cannot create API key with invalid permission
+	req.Permissions = []string{"publisher", "subscriber", "notarealpermission", "topics:create"}
+	_, err = s.client.APIKeyCreate(ctx, req)
+	s.CheckError(err, 400, "validation error: invalid permission specified for apikey")
```

Comment:
Here are the tests for [SC-15075](https://app.shortcut.com/rotational/story/15075)